phoenix-core/src/main/java/org/apache/phoenix/coprocessor/GlobalIndexRegionScanner.java

```diff
@@ -263,7 +263,19 @@ public ImmutableBytesWritable getLatestValue(ColumnReference ref, long ts) {
             valuePtr.set(cell.getValueArray(), cell.getValueOffset(), cell.getValueLength());
             return valuePtr;
         }
-
+        @Override
+        public KeyValue getLatestKeyValue(ColumnReference ref, long ts) {
+            List<Cell> cellList = put.get(ref.getFamily(), ref.getQualifier());
+            if (cellList == null || cellList.isEmpty()) {
+                return null;
+            }
+            Cell cell = cellList.get(0);
+            return new KeyValue(cell.getRowArray(), cell.getRowOffset(), cell.getRowLength(),
```

Comment:
return org.apache.hadoop.hbase.KeyValueUtil.ensureKeyValue(cell)?

Reply:
It's deprecated because we're trying to depeicate KeyValue, I think. Hmmm, perhaps better not to use it.

Do you need a KeyValue as output, or would a Cell work as well?

Reply:
Casting Cell as KeyValue does not work in all cases. It triggers an HBase exception.